packages/core/src/Stores/client-store.js

```diff
@@ -1523,18 +1523,32 @@ export default class ClientStore extends BaseStore {
     setDeviceData() {
         // Set client URL params on init
         const url_params = new URLSearchParams(window.location.search);
+        const date_first_contact_cookie = new CookieStorage('date_first_contact');
+        const signup_device_cookie = new CookieStorage('signup_device');
+
+        if (!date_first_contact_cookie.get('date_first_contact')) {
+            date_first_contact_cookie.set(
+                'date_first_contact',
+                this.root_store.coomon.server_time.format('YYYY-MM-DD')
+            );
+        }
+
+        if (!signup_device_cookie.get('signup_device')) {
+            signup_device_cookie.set('signup_device', isDesktopOs() ? 'desktop' : 'mobile');
+        }
+
         const device_data = {
             ...(url_params.get('affiliate_token') && { affiliate_token: url_params.get('affiliate_token') }),
             ...(url_params.get('gclid_url') && { gclid_url: url_params.get('gclid_url') }),
 
             // date_first_contact should be preserved to the first client contact
             ...(!this.device_data.date_first_contact && {
```

Comment:
```
            ...(url_params.get('date_first_contact') ? { date_first_contact: url_params.get('date_first_contact') } : { date_first_contact: date_first_contact_cookie.get('date_first_contact') }

```

Reply:
I edited this. Please check if it is okay so i can mark it as resolved 👍